src/cocotb/task.py

```diff
@@ -93,13 +104,15 @@ def _get_coro_stack(self) -> Any:
     def __repr__(self) -> str:
         coro_stack = self._get_coro_stack()
 
-        if cocotb._scheduler_inst._current_task is self:
+        if self._state is Task._State.RUNNING:
             fmt = "<{name} running coro={coro}()>"
-        elif self.done():
+        elif self._state is Task._State.FINISHED:
             fmt = "<{name} finished coro={coro}() outcome={outcome}>"
-        elif self._trigger is not None:
+        elif self._state is Task._State.PENDING:
             fmt = "<{name} pending coro={coro}() trigger={trigger}>"
-        elif not self._started:
+        elif self._state is Task._State.SCHEDULED:
+            fmt = "<{name} scheduled coro={coro}()>"
+        elif self._state is Task._State.UNSTARTED:
             fmt = "<{name} created coro={coro}()>"
         else:
```

Comment:
Need a repr for cancelled that should include the `CancelledError`'s message and probably the Task outcome.